docs/testing-plan-guide.md

```diff
@@ -0,0 +1,107 @@
+
+List the various success cases that result in a normal completion:
+
+- Identify algorithm branches, and include a potential input that leads to each branch case.
+- See the proposal's explainer for particular cases that this proposal wants to cover.
+
+#### Failure cases
+
+List failure cases, when an exception is thrown:
+
+- Check new spec text for every place where a `?` appears, indicating that what follows may throw an exception.
+- List other potential unexpected edge cases. At the rendered spec, use the `U` key to see who calls the related parts of the code, and check for unexpected entry points.
+
+#### Side-effects and algorithm step order
+
+Finally, consider testing the order of the algorithm steps, if there are steps that introduce side effects. However, it's best to not overdo it here, the goal is not to have 100% coverage, but to help implementations correctly implement the new functionality.
+
+### Tests for new syntax
+
+A testing plan for new syntax is similar to that for a new API but additionally:
+
+- Test both the syntax and how it's evaluated, so double the tests.
+- Check [CONTRIBUTING.md](https://github.com/tc39/test262/blob/main/CONTRIBUTING.md) for special frontmatter flags when testing syntax errors.
+- Look into the ECMA-262 sections named [Runtime Semantics: Evaluation](https://tc39.es/ecma262/#sec-array-initializer-runtime-semantics-evaluation).
+
+## Tips and tricks
+
+Check what engines test, in case you missed something. Some engine's test locations in their codebase:
+
+- v8 https://source.chromium.org/chromium/chromium/src/+/main:v8/test/mjsunit/
+- SpiderMonkey: https://searchfox.org/mozilla-central/source/js/src/tests/non262/ and https://searchfox.org/mozilla-central/source/js/src/jit-test/tests (note that the latter is usually more focused on testing the JIT compiler, however it may contain more general API tests that are not yet isolated as potential test262 tests).
+- JSC https://trac.webkit.org/browser/webkit/trunk/JSTests
+
+Finally, if you don't understand why a certain part of the spec is the way it is, it helps to git blame the file `spec.html` of the proposal's Git repository. You may find linked PRs pointing to related issues, potentially with test cases that cover that part of the spec.
+
+### Examples of input categories
+
+- `undefined`, `null`, symbols
+- strings, "", "\b", "\n"
+- Booleans
+- number, BigInt, NaN, ±0, ±∞
+- Objects, Arrays
+
+## Completing the testing plan
+
+As important as it is to have a complete testing plan, it is equally important to avoid diminishing returns and when to stop adding cases to be tested. Especially when new specification algorithms call on existing algorithms, it may be temping to dive in the existing algorithms to explore the different paths they may take. However it is best to focus on writing more surface tests for these cases, and only dive in the nested algorithms when the proposal is expecting these to be relevant to the new additions.
```

Comment:
```suggestion
As important as it is to have a complete testing plan, it is equally important to avoid diminishing returns and to understand when to stop adding cases to be tested. Especially when new specification algorithms call on existing algorithms, it may be temping to dive in the existing algorithms to explore the different paths they may take. However it is best to focus on writing more surface tests for these cases, and only dive in the nested algorithms when the proposal is expecting these to be relevant to the new additions.
```
or something like it